controllers/topics/checkName.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    let response = {
+      success: false,
+      message: 'A community with this name already exists',
+      available: false
+    };
+
+    if (topics) {
+      response = {
+        success: true,
+        message: 'Community name available',
+        available: true
+      };
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

-    let response = {
-      success: false,
-      message: 'A community with this name already exists',
-      available: false
-    };
-
-    if (topics) {
-      response = {
-        success: true,
-        message: 'Community name available',
-        available: true
-      };
-    }

+    let response = {
+      success: !topics,
+      message: topics ? 'A community with this name already exists' : 'Community name available',
+      available: !topics
+    };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->